packages/react-reconciler/src/ReactFiberThrow.old.js

```diff
@@ -297,76 +297,82 @@ function throwException(
           wakeables.add(wakeable);
         }
 
-        // If the boundary is in legacy mode, we should *not*
-        // suspend the commit. Pretend as if the suspended component rendered
-        // null and keep rendering. In the commit phase, we'll schedule a
-        // subsequent synchronous update to re-render the Suspense.
-        //
-        // Note: It doesn't matter whether the component that suspended was
-        // inside a concurrent mode tree. If the Suspense is outside of it, we
-        // should *not* suspend the commit.
-        //
-        // If the suspense boundary suspended itself suspended, we don't have to
-        // do this trick because nothing was partially started. We can just
-        // directly do a second pass over the fallback in this render and
-        // pretend we meant to render that directly.
-        if (
-          (workInProgress.mode & ConcurrentMode) === NoMode &&
-          workInProgress !== returnFiber
-        ) {
-          workInProgress.flags |= DidCapture;
-          sourceFiber.flags |= ForceUpdateForLegacySuspense;
+        if ((workInProgress.mode & ConcurrentMode) === NoMode) {
+          // Legacy Mode Suspense
+          //
+          // If the boundary is in legacy mode, we should *not*
+          // suspend the commit. Pretend as if the suspended component rendered
+          // null and keep rendering. When the Suspense boundary completes,
+          // we'll do a second pass to render the fallback.
+          if (workInProgress === returnFiber) {
```

Comment:
This is the special branch that I moved. The rest of the diff is updating comments that referred to outdated implementation details.